js/binance.js

```diff
@@ -6053,6 +6054,57 @@ module.exports = class binance extends Exchange {
         };
     }
 
+    async createMarginLoan (code, amount, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const currency = this.currency (code);
+        const request = {
+            'asset': currency['id'],
+            'amount': amount,
+        };
+        const defaultMargin = this.safeString (params, 'marginMode', 'cross'); // cross or isolated
+        const marginMode = this.safeString2 (this.options, 'defaultMarginMode', 'marginMode', defaultMargin);
+        if (marginMode === 'isolated') {
+            if (symbol === undefined) {
+                throw new BadRequest (this.id + 'createMarginLoan() requires a symbol argument for isolated margin');
+            }
+            request['isIsolated'] = 'TRUE';
+            request['symbol'] = market['id'];
+        }
+        const response = await this.sapiPostMarginLoan (this.extend (request, params));
+        //
+        //     {
+        //         "tranId": 108988250265,
+        //         "clientTag":""
+        //     }
+        //
+        const transaction = this.parseMarginLoan (response, currency['id']);
+        return this.extend (transaction, {
+            'amount': amount,
+            'currency': currency['id'],
+        });
+    }
+
+    parseMarginLoan (info, currency = undefined) {
+        //
+        //     {
+        //         "tranId": 108988250265,
+        //         "clientTag":""
+        //     }
+        //
+        return {
+            'id': this.safeInteger (info, 'tranId'),
+            'currency': undefined,
+            'amount': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'info': info,
+        };
+    }
```

Comment:
We should process `currency` here:

```suggestion
    parseMarginLoan (info, currency = undefined) {
        //
        //     {
        //         "tranId": 108988250265,
        //         "clientTag":""
        //     }
        //
        return {
            'id': this.safeInteger (info, 'tranId'),
            'currency': this.safeCurrencyCode (undefined, currency), // <--------
            'amount': undefined,
            // if the symbol may be an argument of createMarginLoan
            // then should it also be a part of the unified margin loan structure?
            // 'symbol': ? // we probably need it here, even if it's undefined in certain cases
            'timestamp': undefined,
            'datetime': undefined,
            'info': info,
        };
    }
```